components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -666,6 +673,56 @@ internal sealed class SubscriptionWrapper : ISubscription {
                 return subscription;
             }
 
+            private void OnSubscriptionEventChanged(Subscription subscription, EventNotificationList notification, IList<string> stringTable) {
+                try {
+                    if (OnSubscriptionEventChange == null) {
+                        return;
+                    }
+                    if (notification == null) {
+                        _logger.Warning(
+                            "EventChange for subscription: {Subscription} having empty notification",
+                            subscription.DisplayName);
+                        return;
+                    }
+
+                    if (_currentlyMonitored == null) {
+                        _logger.Information(
+                            "EventChange for subscription: {Subscription} having no monitored items yet",
+                            subscription.DisplayName);
+                        return;
+                    }
+
+                    // check if notification is a keep alive
+                    var isKeepAlive = notification?.Events?.Count == 1 &&
+                                      notification?.Events?.First().ClientHandle == 0 &&
```

Comment:
Reading this I'm not sure First() is what was intended. It could be FirstOrDefault(). Given all the redundant null propagation and GetValueOrDefault - I'm getting the sense that it should FirstOrDefault(). First() will simply throw so the null propagation after that is useless.

Reply:
![image](https://user-images.githubusercontent.com/2373875/111977800-74763a00-8b03-11eb-8d01-53d515d23702.png)